File: api/Preferred.Api/Models/SystemResource.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Preferred.Api.Models
{
    /// <summary>
    /// 系统资源占用实体模型
    /// </summary>
    [Table("Tb_SystemResource")]
    public class SystemResource
    {
        /// <summary>
        /// 唯一标识
        /// </summary>
        [Key]
        public int Id { get; set; }

        /// <summary>
        /// 主机名称
        /// </summary>
        [Required]
        [MaxLength(100)]
        public string HostName { get; set; } = string.Empty;

        /// <summary>
        /// CPU占用百分比，保留两位小数
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(5,2)")]
        public decimal CpuUsage { get; set; }

        /// <summary>
        /// 内存占用百分比，保留两位小数
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(5,2)")]
        public decimal MemoryUsage { get; set; }

        /// <summary>
        /// 系统盘名称
        /// </summary>
        [Required]
        [MaxLength(100)]
        public string DiskName { get; set; } = string.Empty;

        /// <summary>
        /// 磁盘占用百分比，保留两位小数
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(5,2)")]
        public decimal DiskUsage { get; set; }

        /// <summary>
        /// 总磁盘容量，单位GB
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal DiskTotal { get; set; }

        /// <summary>
        /// 已使用磁盘容量，单位GB
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal DiskUsed { get; set; }

        /// <summary>
        /// 可用磁盘容量，单位GB
        /// </summary>
        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal DiskFree { get; set; }

        /// <summary>
        /// 创建时间
        /// </summary>
        [Required]
        public DateTime CrtTime { get; set; } = DateTime.Now;

        /// <summary>
        /// 最后修改时间
        /// </summary>
        [Required]
        public DateTime UpdTime { get; set; } = DateTime.Now;
    }
}

File: ui/preferred-admin/src/api/systemResource.ts
import request from '@/utils/request'

// 系统资源相关的接口类型定义
export interface SystemResource {
  id: number
  hostName: string
  cpuUsage: number
  memoryUsage: number
  diskName: string
  diskUsage: number
  diskTotal: number
  diskUsed: number
  diskFree: number
  crtTime: string
  updTime: string
}

export interface SystemResourceStats {
  avgCpuUsage: number
  maxCpuUsage: number
  avgMemoryUsage: number
  maxMemoryUsage: number
  avgDiskUsage: number
  maxDiskUsage: number
}

export interface ApiResponse<T> {
  success: boolean
  message: string
  data: T
}

// 系统资源管理API
export const systemResourceApi = {
  // 获取最新的系统资源信息
  getLatestSystemResource(): Promise<ApiResponse<SystemResource>> {
    return request.get('/api/systemresource/latest')
  },

  // 获取一天内的系统资源数据
  getDailySystemResourceData(): Promise<ApiResponse<SystemResource[]>> {
    return request.get('/api/systemresource/daily')
  },

  // 获取系统资源统计信息
  getSystemResourceStats(): Promise<ApiResponse<SystemResourceStats>> {
    return request.get('/api/systemresource/stats')
  }
}

export default systemResourceApi